gtime/hierarchical/bottom_up.py

```diff
@@ -0,0 +1,137 @@
+from gtime.hierarchical import HierarchicalNaive
+from typing import Union, Dict, Any, List
+import pandas as pd
+from sklearn.base import BaseEstimator
+from sklearn.utils.validation import check_is_fitted
+from copy import deepcopy
+
+
+
+class BottomUp(HierarchicalNaive):
+    """ Hierarchical model with prediction following the Bottom-Up procedure.
+    It computes the forecast of the leaves in the hierarchical tree structure
+    and it performs the sums of these forecasts to recover the prediction on the
+    higher level.
+
+    Parameters
+    ----------
+    model: BaseEstimator, required
+        time series forecasting model that is applied to each of the time series. A cross validation model
+        can also be passed.
+    hierarchy_tree: String or Dict[str, List], optional, default: 'Infer'
+        Adjacency list to import the structure of the hierarchical tree. Keys must be the same as in the
+        dictionary X containing the dataframes. The items are lists containing the name of the children
+        of each key. In case there are no children (the key is a leaf) the item must be an empty list.
+    Examples
+    --------
+    >>> import pandas._testing as testing
+    >>> from gtime.time_series_models import AR
+    >>> from gtime.hierarchical import BottomUp
+    >>>
+    >>> testing.N, testing.K = 20, 1
+    >>> data1 = testing.makeTimeDataFrame(freq="s")
+    >>> data2 = testing.makeTimeDataFrame(freq="s")
+    >>> data3 = testing.makeTimeDataFrame(freq="s")
+    >>> data = {'data1': data1, 'data2': data2, 'data3' : data3}
+    >>> time_series_model = AR(p=2, horizon=3)
+    >>> tree = {'data1' : ['data2', 'data3'], 'data2': [], 'data3': []}
```

Comment:
Maybe you can consider a networkx DiGraph as graph type (or any structure that can be converted into that)